lib/tasks/dad_tasks.rake

```diff
@@ -244,12 +244,18 @@ namespace :dad_tasks do
       (1..3).each do |i|
         House.create(
           house_name: "House #{i}",
-          house_start: Time.now.utc.at_beginning_of_month.to_date,
+          house_start: Time.now.utc.at_beginning_of_month.to_date
         )
       end
     end
 
-    # STEP 7: Now that the daily job is complete, update.
+    # STEP 8 Update any users that are not approved at level 3
+    User.where('highest_level >= 3 AND approved = false').each do |u|
+      u.approve("User #{u.name} (id=#{u.id}) was automatically approved for reaching level"\
+                " #{u.highest_level}.", User.where(is_admin: true).first)

```

Comment:
Hmm. While I would love to be flagged as an awesome robot what is responsible for approving the users (also, glad we actually have a class method for this), I think maybe we should probably adjust moderator_logs and relevant pages to interpret a user_id = 0 as being 'DADBot' or something. This may be more of an extensive fix, but it's not exactly accurate to say this was an action taken by me.
Depending on how annoying that is, we can just leave it as this for the time being. There is a danger point in the rollover script breaking, which is less relevant for the site, but more relevant for, say, QA and dev environments.
Another alternative is to just do it dirty, with an update_attribute method, but that doesn't leave a trace. That said, I don't think it really matters?

Reply:
The only thing is the User.approve method needs an active record to a user, not just an ID. Since there is no User.find(0) it the User.approve method would break.

I think this method works and there is at least a paper trail of what happened. Your call I can change it if you want.